src/render.js

```diff
@@ -32,20 +31,67 @@ const resetGlobals = function() {
 
 resetGlobals();
 
-
 const runInContext = function(context, fn) {
     return vm.runInContext("(" + fn.toString() + ")()", context);
 };
 
+/**
+ * This method is executed whenever a render is needed. It is executed inside
+ * the vm context.
+ */
+const performRender = async () => {
```

Comment:
This is the main method we execute inside the client context once the entry point is loaded.